kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/zookeeper/DiscoveryClient.scala

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.ha.client.zookeeper
+
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.TimeUnit
+
+import scala.collection.JavaConverters._
+
+import com.google.common.annotations.VisibleForTesting
+import org.apache.curator.framework.CuratorFramework
+import org.apache.curator.framework.recipes.locks.InterProcessSemaphoreMutex
+import org.apache.curator.framework.recipes.nodes.PersistentNode
+import org.apache.curator.utils.ZKPaths
+import org.apache.zookeeper.CreateMode
+
+import org.apache.kyuubi.KyuubiSQLException
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.ha.client.KyuubiDiscoveryClient
+import org.apache.kyuubi.ha.client.ServiceNodeInfo
+import org.apache.kyuubi.ha.client.zookeeper.ClientProvider.buildZookeeperClient
+import org.apache.kyuubi.ha.client.zookeeper.ServiceDiscoveryClient.createServiceNode
+
+class DiscoveryClient(conf: KyuubiConf) extends KyuubiDiscoveryClient {
```

Comment:
Can we name the trait `DiscoveryClient` and the derived class `ZookeeperDiscoveryClient`/`EtcdDiscoveryClient` as the proposal documented?

Reply:
Sure, had rename.